src/components/layouts/articleAdministration/ProductOverview.js

```diff
@@ -0,0 +1,108 @@
+import React, {useEffect, useState} from 'react';
+import ProductTable from "./productTable";
+import Button from "@material-ui/core/Button";
+import * as API from "../../connectionHandler/connectionHandler";
+import AddProductDialog from "./addProductDialog";
+import AddNpkProductDialog from "./addNpkProductDialog";
+import {makeStyles} from "@material-ui/core/styles";
+
+
+const ProductOverview = ({ setErrorMessage, customers, onCancel, onSubmit, show, ...props }) =>  {
+
+  const [showNewProductDialog, setNewProductDialogViewState] = useState(false);
+  const [showNewNpProductDialog, setNewNpkProductDialogViewState] = useState(false);
+  const [productData, setProductData] = useState([]);
+  const buttonName = "Neuen Artikel hinzufügen";
+  const buttonNpkName = "Neuen NPK Artikel hinzufügen";
+  const [products, setProducts] = useState([]);
+  const [npks, setNpks] = useState([]);
+
+  const useStyles = makeStyles({
+    overlay:{
+      top: '50px',
+      width: '100%',
+      backgroundColor: 'white',
+      position: "fixed",
+      height: "100%",
+      bottom: "50px",
+      zIndex: 999,
+      padding: "20px",
+      boxSizing: 'border-box'
+    },
+    button:{
+      marginRight: '10px',
+      marginBottom: '10px'
+    }
+  });
+
+  const classes = useStyles();
+
+  const closeNewProductDialog = () => {
+    setNewProductDialogViewState(false);
+    setProductData([]);
+  }
+
+  const closeNewNpkProductDialog = () => {
+    setNewNpkProductDialogViewState(false);
+    setProductData([]);
+  }
+
+  const getProducts = () => {
+    API.getProducts(setErrorMessage, setProducts);
+  }
+
+  const loadNewProducts = () => {
+    closeNewProductDialog();
+    closeNewNpkProductDialog();
+    getProducts();
+  }
+
+  const openNewProductDialog = () => {
+    setNewProductDialogViewState(true);
+  }
+
+  const openNewNpkProductDialog = () => {
+    setNewNpkProductDialogViewState(true);
+  }
+
+/*  const submitNewArticle = (newArticleData) => {
+    setNewProductDialogViewState(false);
+    console.log(JSON.stringify(newArticleData));
+    API.submitNewProduct(newArticleData, setErrorMessage, getProducts);
+  }*/
+
+  const addNewProductDialog =
+      <AddProductDialog
+          show={showNewProductDialog}
+          articles={productData}
+          onCancel={closeNewProductDialog}
+          onSubmit={loadNewProducts}
+          products={products}
+          npks={npks}
+          setNpks={setNpks}
+      />
+
```

Comment:
these 2 dialogs appear to do the same? Or why do both take the same parameters? If my understanding is right, one creates articles without NPK (but still has npks as parameter?).

Reply:
your right, they are gone